### DISCUSSION TITLE
Can i use input filter type as argument?

### DISCUSSION THREAD

**Discussion by john2094** [ChilliCream/hotchocolate#4536] [Q&A]
It's posible use a where input as argument of query/mutation?. I need apply this filter before resolvers done. 

Something like this:

```
        public async UsersMutationResult UpdateUsers(UpdateUsersInput data, [GraphQLType(typeof(FilterInputType<User>))] object where)
        {
            /// here I need apply where filter input
        }
```

Another example is how the @neo4j/graphql library does it with its mutations to update, they accept **where** as a parameter.

https://neo4j.com/docs/graphql-manual/current/mutations/update/

cheers

**Comment by zliebersbach** [comment]
Edit: @john2094 I found a solution!

You can use the `UseFiltering` attribute on a mutation and then call the `Filter` extension method on `IQueryable` to apply the filter from the context. See example below!

```c#
[UseFiltering(typeof(Sample))] // This sets up the filtering
public async Task<IUpdateSamplesResult?> UpdateSamples(Guid id,
        UpdateSampleInput input,
        IResolverContext resolverContext, // Required to read filter context
        [Service] DbShardingManager dbShardingManager)
{
    await using var dbContext = dbShardingManager.ResolveDbContext(id);
    var query = await dbContext.Samples.Filter(resolverContext).ToListAsync(); // This applies the filter from the resolverContext

    // rest of resolver code here
}
```

**Reply by john2094** [reply]
Hello!, No, I did not find a solution. I spent a lot of time understanding how the usefilter attribute works and how hot chocolate builds a query, but to no avail. Finally, due to the delivery deadlines of the product I'm working on, I had to put this idea aside, and make my mutation receive a list of ids instead of a query.

**Reply by zliebersbach** [reply]
@john2094 See my edited response! This solves the issue for me, hope you find it useful. 😄

**Reply by john2094** [reply]
@zliebersbach, this is Awesone! 🎉. Thanks.
